setuptools/dist.py

```diff
@@ -145,11 +145,11 @@ def read_pkg_file(self, file):
 
 
 def single_line(val):
-    # quick and dirty validation for description pypa/setuptools#1390
+    """Validate that the value does not have line breaks."""
+    # Ref: https://github.com/pypa/setuptools/issues/1390
     if '\n' in val:
-        # TODO after 2021-07-31: Replace with `raise ValueError("newlines not allowed")`
-        warnings.warn("newlines not allowed and will break in the future")
-        val = val.replace('\n', ' ')
+        raise ValueError('Newlines are not allowed')
```

Comment:
The problem with this wording is that it doesn't help figure out what's wrong. When I saw a warning in the middle of the console output, the only way to understand what it was, was to read the source code. This is not the best UX, is it?

Reply:
Indeed, it's not the best user experience. But I expect these errors to be rare. Also, now that they're raised as an exception (or if you'd passed `-We`), you'll see the traceback, which does point to `single_line(self.get_description())`, which does indicate where the error is.